progs/tests/expression/if_expressions.sy

```diff
@@ -0,0 +1,24 @@
+grouping :: fn {
+    a := 1 + 2 + 3 if true else -1 * 10 if false else 2 + 2
+    a <=> 4
+    b := 1 + 2 + 3 if true else -1 * 10 if true else 2 + 2
+    b <=> 6
+    c := 1 + 2 + 3 if false else -1 * 10 if true else 2 + 2
+    c <=> -10
+    d := 1 + 2 + 3 if false else -1 * 10 if false else 2 + 2
+    d <=> 4
+}
```

Comment:
I actually don't agree with this grouping. I read these like this (parenthesis added for clarity):

```
    a := 1 + 2 + (3 if true else (-1 * (10 if false else (2 + 2))))
    a <=> 6
    b := 1 + 2 + (3 if true else (-1 * (10 if true else (2 + 2))))
    b <=> 6
    c := 1 + 2 + (3 if false else (-1 * (10 if true else (2 + 2))))
    c <=> -7
    d := 1 + 2 + (3 if false else (-1 * (10 if false else (2 + 2))))
    d <=> -1
```
I think this is clearer since here, if the condition is true, you know that the rest of the expression isn't necessary. (Unless you have parenthesis.)

Reply:
I don't know how nice this is to read - when they are nested. But I like the clarity that you get - I don't expect many people to nest these after eachother. 

But why don't you want this grouping:
```
(3 if true else -1) * (10 if false else 2) + 2
```
I feel like the binding rules you propose are inconsistent - since they seam to have a "rest of the line" kind of semantic to them.

Why is the first 3 cut of but not the last 2? If I were the first 3 I would be pissed!

Either everyone gets a buddy - or noone gets a buddy! >:(